src/setfit/modeling.py

```diff
@@ -239,6 +245,8 @@ def fit(
                     # to model's device
                     features = {k: v.to(device) for k, v in features.items()}
                     labels = labels.to(device)
+                    if self.model_head.multitarget:
+                        labels = labels.float()
```

Comment:
Added this while waiting for #187. Required for the loss function using multitarget labels

Reply:
I will suggest this PR be merged before #203 and #187.
Please see the reasons [here](https://github.com/huggingface/setfit/issues/211#issuecomment-1331656303) in #211.